judgments/views/detail.py

```diff
@@ -79,23 +84,26 @@ def detail(request, document_uri):
     context = {}
 
     press_summary_suffix = "/press-summary/1"
-    if document_uri.endswith(press_summary_suffix):
-        context["document_type"] = "press_summary"
+    context["document_noun"] = document.document_noun
+    if document.best_human_identifier is None:
+        raise NoNeutralCitationError(document.uri)
+    context["judgment_ncn"] = document.best_human_identifier
+
+    if document.document_noun == "press summary":
         context["linked_document_uri"] = document_uri.removesuffix(press_summary_suffix)
-        context["judgment_ncn"] = ""
     else:
-        context["document_type"] = "judgment"
         context["linked_document_uri"] = document_uri + press_summary_suffix
-        context["judgment_ncn"] = document.neutral_citation
 
-    linked_document = None
     try:
         linked_document = get_published_document_by_uri(context["linked_document_uri"])
     except Http404:
         context["linked_document_uri"] = ""
-
-    if context["document_type"] == "press_summary" and linked_document:
-        context["judgment_ncn"] = linked_document.neutral_citation
+    else:
+        if context["document_noun"] == "press summary" and linked_document:
+            if linked_document.best_human_identifier:
```

Comment:
On lone 88 we check for `if document.best_human_identifier is None:`

can we make these 2 checks consistent?

Reply:
I probably wouldn't go as far as verifying the ncn is valid, but that does make me think again that we have validation in EUI to block publishing if ncn is invalid in any way, so maybe we could assume that best_human_identifier is valid for judgments and press summaries in the PUI and so NoNeutralCitationError is unnecessary?